delta-lake/delta-spark332db/src/main/scala/com/nvidia/spark/rapids/delta/shims/DeltaLogShim.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.delta.shims
+
+import com.databricks.sql.transaction.tahoe.DeltaLog
+import com.databricks.sql.transaction.tahoe.actions.Metadata
+import com.databricks.sql.transaction.tahoe.commands.{MergeIntoCommand, MergeIntoCommandEdge}
+import com.nvidia.spark.rapids.delta.{MergeIntoCommandEdgeMeta, MergeIntoCommandMeta}
+
+import org.apache.spark.sql.execution.datasources.FileFormat
+
+object DeltaLogShim {
+  def fileFormat(deltaLog: DeltaLog): FileFormat = {
+    deltaLog.fileFormat(deltaLog.unsafeVolatileSnapshot.metadata)
+  }
+  def getMetadata(deltaLog: DeltaLog): Metadata = {
+    deltaLog.unsafeVolatileSnapshot.metadata
+  }
+
+  def tagForGpu(meta: MergeIntoCommandMeta, mergeCmd: MergeIntoCommand): Unit = {
+    // see https://github.com/NVIDIA/spark-rapids/issues/8415 for more information
+    if (mergeCmd.notMatchedBySourceClauses.nonEmpty) {
+      meta.willNotWorkOnGpu("notMatchedBySourceClauses not supported on GPU")
+    }
+  }
+
+  def tagForGpu(meta: MergeIntoCommandEdgeMeta, mergeCmd: MergeIntoCommandEdge): Unit = {
+    // see https://github.com/NVIDIA/spark-rapids/issues/8415 for more information
+    if (mergeCmd.notMatchedBySourceClauses.nonEmpty) {
+      meta.willNotWorkOnGpu("notMatchedBySourceClauses not supported on GPU")
+    }
+  }
```

Comment:
The classname implies this is a shim for DeltaLog, but these methods have nothing to do with that class. There should be a shim for MergeIntoCommandMeta[Edge] since that's what this relates to (typically by leveraging a MergeIntoCommandMetaBase that has all the common stuff and 332db would have extra stuff).  Or maybe this class should be named DeltaLakeShim or something to be more generic and become a catch-all shim for Delta Lake stuff.